pypeit/core/pydl.py

```diff
@@ -1851,26 +1852,39 @@ def djs_reject(data, model, outmask=None, inmask=None, sigma=None,
     # of multiples of maxdev away from the fit.
     #
     badness = np.zeros(outmask.shape, dtype=data.dtype)
+
+    if percentile:
+        if inmask is not None:
+            igood = (inmask & outmask)
+        else:
+            igood = outmask
+        if (np.sum(igood)> 1):
+            if lower is not None:
+                lower_chi = np.percentile(chi[igood],lower)
+            else:
+                lower_chi = -np.inf
+            if upper is not None:
+                upper_chi = np.percentile(chi[igood], upper)
+            else:
+                upper_chi = np.inf
```

Comment:
```
    if percentile:
        igood = outmask
        if inmask is not None:
            igood &= inmask
        if np.any(igood):
            _lower = -np.inf if lower is None else np.percentile(chi[igood],lower)
            _upper = np.inf if upper is None else np.percentile(chi[igood], upper)
    else:
        _lower = None if lower is None else -lower
        _upper = upper

    # Decide how bad a point is according to lower.
    if _lower is not None:
        badness += np.fmax(-chi,0.0)*(chi < _lower)

    # Decide how bad a point is according to upper.
    if _upper is not None:
        badness += np.fmax(chi,0.0)*(chi > _upper)
```

Reply:
Go ahead and feel free to implement this change in a future short PR. I'm reluctant to fiddle with this routine right now without carefully testing it, as it is a core routine. It should in fact be in the test suite, but is not yet.